dcp/api/job_fs.py

```diff
@@ -0,0 +1,234 @@
+"""
+Job Filesystem.
+
+A vfsystem to be used for dcp jobs via tar vfs.
+
+Note: Currently only supported in the Pyodide worktime.
+Note: This logic will be mostly replaced by dcp-client's JobFS when it lands.
+
+Author: Will Pringle <will@distributive.network>
+Date: July 2024
+"""
+import pathlib
+import tarfile
+import os
+import io
+
+import readline
+import atexit
+import rlcompleter
+
+class JobFS:
+    """Job Virtual Filesystem (JobFS)."""
+    def __init__(self):
+        self.vfs = {}
+        self.vfs['/'] = self.vfs
+        self.home = '/home/pyodide'
+        self.mkdir(self.home)
+        self.cwd = self.home
+
+    def add(self, local_src, vfs_dest=None):
+        """Adds local file[s] to the vfs.
+        If dest not specified, add file from root.
+        Overwrites file if there is already one specified there.
+        local_src can be of type:
+            - string        : file path 
+            - pathlib.Path  : file path
+            - bytes         : data blob
+            - bytearray     : data blob
+            - file-like     : data blob
+        vfs_dest can be of type:
+            - string        : file path (optional)
+            - pathlib.Path  : file path (optional)
+        vfs_dest is not optional when passing data blobs.
+        """
+        if isinstance(local_src, io.IOBase):
+            local_src = local_src.read()
+
+        if isinstance(local_src, bytes) or isinstance(local_src, bytearray):
+            if vfs_dest is None:
+                raise Exception('Must specify a destination file path to write to')
+        elif isinstance(local_src, str) or isinstance(local_src, pathlib.PurePath):
+            if vfs_dest is None:
+                vfs_dest = os.path.join(self.home, os.path.basename(local_src))
```

Comment:
Probably should be added from the current working directory, not home

Reply:
0a46495e06ddee99f35a159b9569b36a04b02be2